blocks/procedures.ts

```diff
@@ -555,36 +578,47 @@ blocks['procedures_mutatorcontainer'] = {
             fieldRegistry.fromJson({
               type: 'field_checkbox',
               checked: true,
-            }),
+            }) as FieldCheckbox,
             'STATEMENTS');
     this.setStyle('procedure_blocks');
     this.setTooltip(Msg['PROCEDURES_MUTATORCONTAINER_TOOLTIP']);
     this.contextMenu = false;
   },
 };
+blocks['procedures_mutatorcontainer'] = PROCEDURES_MUTATORCONTAINER;
 
-blocks['procedures_mutatorarg'] = {
-  /**
+/** Type of a procedures_mutatorarg block. */
+type ArgumentBlock = Block&ArgumentMixin;
+interface ArgumentMixin extends ArgumentMixinType {}
+type ArgumentMixinType = typeof PROCEDURES_MUTATORARGUMENT;
+
+// TODO(#6920): This is kludgy.
+type FieldTextInputForArgument = FieldTextInput & {
+  oldShowEditorFn_(_e?: Event, quietInput?: boolean): void,
+  createdVariables_: VariableModel[],
+};
+
+const PROCEDURES_MUTATORARGUMENT = {
+    /**
    * Mutator block for procedure argument.
-   * @this {Block}
    */
-  init: function() {
+  init: function(this: ArgumentBlock) {
     const field = fieldRegistry.fromJson({
       type: 'field_input',
       text: Procedures.DEFAULT_ARG,
-    });
+    }) as FieldTextInputForArgument;
     field.setValidator(this.validator_);
     // Hack: override showEditor to do just a little bit more work.
     // We don't have a good place to hook into the start of a text edit.
-    field.oldShowEditorFn_ = field.showEditor_;
+    field.oldShowEditorFn_ = (field as AnyDuringMigration).showEditor_;
```

Comment:
Maybe add a TODO comment pointing at #6920  for this?

Reply:
I could, but #6920 already mentions use of `as`, and I had been treating the "DuringMigration" part as making it implicitly clear that it was not OK.

I've  added a note to #6920 to check for `AnyDuringMigration` in particular, though.